src/components/PeopleTable.tsx

```diff
@@ -1,645 +1,106 @@
+      const matchCentury = centuries.length > 0 ? centuries.includes(numberCentury.toString()) : true;
+
+      return matchesSex && matchesName && matchCentury;
+    }) || [];
+
+    setLocal(filteredData);
+  }, [sex, nameSearch, peopleData, centuries]);
+
   return (
     <table
       data-cy="peopleTable"
       className="table is-striped is-hoverable is-narrow is-fullwidth"
     >
-      <thead>
-        <tr>
-          <th>
-            <span className="is-flex is-flex-wrap-nowrap">
-              Name
-              <a href="#/people?sort=name">
-                <span className="icon">
-                  <i className="fas fa-sort" />
+      {local?.length === 0 ? (
+        <caption>There are no people matching the current search criteria</caption>
+      ) : (
+        <>
+          <thead>
+            <tr>
+              <th>
+                <span className="is-flex is-flex-wrap-nowrap">
+                  Name
+                  <a>
+                    <span className="icon">
+                      <i className="fas fa-sort" />
+                    </span>
+                  </a>
                 </span>
-              </a>
-            </span>
-          </th>
-
-          <th>
-            <span className="is-flex is-flex-wrap-nowrap">
-              Sex
-              <a href="#/people?sort=sex">
-                <span className="icon">
-                  <i className="fas fa-sort" />
+              </th>
+              <th>
+                <span className="is-flex is-flex-wrap-nowrap">
+                  Sex
+                  <a>
+                    <span className="icon">
+                      <i className="fas fa-sort" />
+                    </span>
+                  </a>
                 </span>
-              </a>
-            </span>
-          </th>
-
-          <th>
-            <span className="is-flex is-flex-wrap-nowrap">
-              Born
-              <a href="#/people?sort=born&amp;order=desc">
-                <span className="icon">
-                  <i className="fas fa-sort-up" />
+              </th>
+              <th>
+                <span className="is-flex is-flex-wrap-nowrap">
+                  Born
+                  <a>
+                    <span className="icon">
+                      <i className="fas fa-sort-up" />
+                    </span>
+                  </a>
                 </span>
-              </a>
-            </span>
-          </th>
-
-          <th>
-            <span className="is-flex is-flex-wrap-nowrap">
-              Died
-              <a href="#/people?sort=died">
-                <span className="icon">
-                  <i className="fas fa-sort" />
+              </th>
+              <th>
+                <span className="is-flex is-flex-wrap-nowrap">
+                  Died
+                  <a>
+                    <span className="icon">
+                      <i className="fas fa-sort" />
+                    </span>
+                  </a>
```

Comment:
The sorting icons are present, but there is no implementation for sorting the data when these icons are clicked. Consider adding event handlers to sort the data by `name`, `sex`, `born`, and `died` when the respective sort icons are clicked.